qemu/tests/driver_in_use.py

```diff
@@ -0,0 +1,125 @@
+import re
+import time
+import logging
+from virttest import utils_misc, utils_test, error_context
+from autotest.client.shared import utils
+from avocado.core import exceptions
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Driver in use test:
+    1) boot guest with the device.
+    2) enable and check driver verifier in guest.
+    3) run subtest before / during / after background stress test.
+    4) clear the driver verifier.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def check_bg_status(session, target):
+        """
+        Check the backgroud test status in guest.
+
+        :param session: VM session.
+        :param target: Driver name.
+        :return: return True if find the driver name;
+                 else return False
+        """
+        list_cmd = "wmic process where name='%s' list" % target
+        try:
+            output = session.cmd_output_safe(list_cmd, timeout=120)
+            check_reg = re.compile(r"%s" % target, re.I | re.M)
+            if bool(check_reg.findall(output)):
+                return True
+        except Exception, err:
+            raise exceptions.TestFail("Check process error: %s" % str(err))
+        return False
+
+    def bg_stress_test():
+        """
+        Run backgroud test.
+        """
+        bg_stress_test = params.get("run_bgstress")
+        if bg_stress_test:
+            try:
+                error_context.context("Run test %s background" % bg_stress_test,
+                                      logging.info)
+                stress_thread = ""
+                wait_time = float(params.get("wait_bg_time", 60))
+                target = params.get("target", "")
+                bg_stress_run_flag = params.get("bg_stress_run_flag")
```

Comment:
Hi shuang,

+                stress_thread = utils.InterruptedThread(
+                    utils_test.run_virt_sub_test, (test, params, env),
+                    {"sub_type": bg_stress_test})

In the background case, the flag will be set.

Refer (from netperf_stress):
            # here when set a run flag, when other case call this case as a
            # subprocess backgroundly, can set this run flag to False to stop
            # the stress test.
            env["netperf_run"] = True